### PR TITLE
Document ways to run vulnerability management

### PR DESCRIPTION
**Pull request by amirbenun**
### Summary of your changes
Document how developers should run vulnerability management flow in two use cases:
1. Using CloudFormation on AWS
2. Locally

### Related Issues
- Resolves: https://github.com/elastic/cloudbeat/issues/833

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: deploy/vulnerability/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+We will do that by running cloudbeat with a custom `cloudbeat.yml` file that triggers the vulnerability management flavor.
+
+### How to run it
+*Prerequisites:*
+1. You have an elasticsearch deployed locally or in the cloud
+2. You have AWS CLI installed on your laptop and configured to work with our dev account `elastic-security-cloud-security-dev`
+
+*Steps:*
+1. From your IDE add the following to your running options (on vscode `launch.json`)
+```
```
Comment (oren-zohar) [inline deploy/vulnerability/README.md:14]:
```suggestion
```json
```

File: deploy/vulnerability/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        "env": {
+                "ES_HOST": "https://localhost:9200",
+                "ES_USERNAME": "elastic",
+                "ES_PASSWORD": "changeme",
+        }
+}
+```
+2. Hit "Launch" to run cloudbeat standalone (no agent) and generate vulnerabilities to your elasticsearch.
+3. Alternatively, you can run
+```
```
Comment (oren-zohar) [inline deploy/vulnerability/README.md:31]:
```suggestion
```shell
```

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: deploy/cloudformation/README.md [hunk 3]
```diff
@@ -5,18 +5,27 @@ This CloudFormation template creates a role for elastic-agent and attaches it to
...
 STACK_NAME="your unique stack name"
 FLEET_URL="<Elastic Agent Fleet URL>"
 ENROLLMENT_TOKEN="<Elastic Agent Enrollment Token>"
 ELASTIC_AGENT_VERSION="<Elastic Agent Version>" # e.g: 8.8.0 | 8.8.0-SNAPSHOT
 
-DEV.ALLOW_SSH=bool # Set to true in order to modify the template to allow SSH connections
-DEV.KEY_NAME="" # When SSH is allowed, your EC2 SSH key is required
-DEV.PRE_RELEASE=bool # Set to true in order to replace the artifact URL with a pre-release version (BC or snapshot)
-DEV.SHA="" # When running a pre-release version, you have to specify the SHA of the pre-release artifact (on SNAPSHOT versions you can leave empty to take the latest)
+DEV.ALLOW_SSH=bool # Set to true to modify the template to allow SSH connections
```
[the remark this answers is not in the document]
Comment (orestisfl) [reply deploy/cloudformation/README.md:22]:
```shell
$ just deploy-cloudformation
cd deploy/cloudformation && go run .
2023/05/01 15:37:22 failed to unmarshal configuration file: 2 error(s) decoding:

* cannot parse 'DEV.ALLOW_SSH' as bool: strconv.ParseBool: parsing "bool": invalid syntax
* cannot parse 'DEV.PRE_RELEASE' as bool: strconv.ParseBool: parsing "bool": invalid syntax
```

File: deploy/cloudformation/README.md [hunk 4]
```diff
@@ -5,18 +5,27 @@ This CloudFormation template creates a role for elastic-agent and attaches it to
...
+DEV.KEY_NAME="" # When SSH is allowed, your EC2 SSH key name is required
+DEV.PRE_RELEASE=bool # Set to true to replace the artifact URL with a pre-release version (BC or snapshot)
+DEV.SHA="" # When running a pre-release version, you have to specify the SHA of the pre-release artifact (on SNAPSHOT versions you can leave it empty to take the latest)
 ```
+4. Run `just deploy-cloudformation` to create a CloudFormation stack with an elastic-agent that will automatically enroll to your fleet.
```
[the remark this answers is not in the document]
Comment (amirbenun) [reply deploy/cloudformation/README.md:27]: Yes exactly.

File: deploy/cloudformation/README.md [hunk 5]
```diff
@@ -5,18 +5,27 @@ This CloudFormation template creates a role for elastic-agent and attaches it to
...
 
-Run `just deploy-cloudformation`
+*Debugging:*
+1. To track the progress of the CloudFormation stack creation go to https://console.aws.amazon.com/cloudformation/
+2. If the stack was created successfully but elastic-agent didn't enroll to your fleet, try to ssh into the EC2 and get the initialization logs by `cat /var/log/cloud-init-output.log`.
```
[the remark this answers is not in the document]
Comment (amirbenun) [reply deploy/cloudformation/README.md]: Added the ssh command

File: deploy/cloudformation/README.md [hunk 6]
```diff
@@ -5,18 +5,27 @@ This CloudFormation template creates a role for elastic-agent and attaches it to
 The EC2 instance has elastic-agent preinstalled in it using the fleet URL and enrollment token.
 
 ### How to test it
-The template can be tested with AWS SDK as follows:
-Create a `.env` file of the form:
+*Prerequisites:*
+1. You have an elastic stack deployed in the cloud that includes Kibana, elasticsearch and fleet-server
+2. You have AWS CLI installed on your laptop and configured to work with our dev account `elastic-security-cloud-security-dev` (in particular, `~/.aws/config` and `~/.aws/credentials` should be set)
```
[the remark this answers is not in the document]
Comment (amirbenun) [reply deploy/cloudformation/README.md]: Added a link

File: deploy/cloudformation/README.md [hunk 7]
```diff
@@ -5,18 +5,27 @@ This CloudFormation template creates a role for elastic-agent and attaches it to
 The EC2 instance has elastic-agent preinstalled in it using the fleet URL and enrollment token.
 
 ### How to test it
-The template can be tested with AWS SDK as follows:
-Create a `.env` file of the form:
+*Prerequisites:*
+1. You have an elastic stack deployed in the cloud that includes Kibana, elasticsearch and fleet-server
```
[the remark this answers is not in the document]
Comment (amirbenun) [reply deploy/cloudformation/README.md]: Added a link

File: deploy/vulnerability/README.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+## Cloudbeat standalone runs vulnerability management
+
+### What it does
+This document explains how to locally run cloudbeat standalone with vulnerability management flow.
+We will do that by running cloudbeat with a custom `cloudbeat.yml` file that triggers the vulnerability management flavor.
+
+### How to run it
+*Prerequisites:*
+1. You have an elasticsearch deployed locally or in the cloud
```
[the remark this answers is not in the document]
Comment (amirbenun) [reply deploy/vulnerability/README.md]: Added a link

File: deploy/vulnerability/README.md [hunk 9]
```diff
@@ -6,7 +6,7 @@ We will do that by running cloudbeat with a custom `cloudbeat.yml` file that tri
 
 ### How to run it
 *Prerequisites:*
-1. You have an elasticsearch deployed locally or in the cloud
+1. You have an elasticsearch deployed locally or in the cloud (check https://github.com/elastic/elastic-package to run elastic stack locally)
```
Comment (oren-zohar) [inline deploy/vulnerability/README.md:9]: I think we can ref our docs here https://github.com/elastic/cloudbeat/blob/main/dev-docs/ELK-Deployment.md

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: deploy/cloudformation/README.md [hunk 10]
```diff
@@ -5,18 +5,27 @@ This CloudFormation template creates a role for elastic-agent and attaches it to
 The EC2 instance has elastic-agent preinstalled in it using the fleet URL and enrollment token.
 
 ### How to test it
-The template can be tested with AWS SDK as follows:
-Create a `.env` file of the form:
+*Prerequisites:*
+1. You have an elastic stack deployed in the cloud that includes Kibana, elasticsearch and fleet-server (check https://cloud.elastic.co/ to deploy your own stack)
```
Comment (oren-zohar) [inline deploy/cloudformation/README.md:9]: https://github.com/elastic/cloudbeat/blob/main/dev-docs/ELK-Deployment.md same here

**Review by oren-zohar — CHANGES_REQUESTED** [review]

File: deploy/cloudformation/README.md [hunk 11]
```diff
@@ -5,18 +5,27 @@ This CloudFormation template creates a role for elastic-agent and attaches it to
...
+2. You have AWS CLI installed on your laptop and configured to work with our dev account `elastic-security-cloud-security-dev` (in particular, `~/.aws/config` and `~/.aws/credentials` should be set, check https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html for more information)
+
+*Steps:*
+1. Install Vulnerability Management integration on a new agent policy, you might have to check the "Display beta integrations" checkbox.
+2. After you installed the integration you can install a new elastic-agent, you should keep the fleet URL and the enrollment token.
+3. On cloudbeat repo, create a `deploy/cloudformation/.env` file of the form:
```
Comment (kfirpeled) [inline deploy/cloudformation/README.md:15]: nit can you please add an example stack name? so others would follow

like
`STACK_NAME="your unique stack name" # john-qa-bc2-8-9-0-May28`

**Review by kfirpeled — APPROVED** [review]

**Review by oren-zohar — APPROVED** [review]
:pray::pray::pray:

**Pull request merged**